addons/website_blog/static/src/snippets/s_blog_posts/options.js

```diff
@@ -12,8 +12,13 @@ const dynamicSnippetBlogPostsOptions = dynamicSnippetOptions.extend({
         this._super.apply(this, arguments);
         this.modelNameFilter = 'blog.post';
         this.blogs = {};
+        this.tags = this.$target[0].dataset.filterByTagIds ? this.$target[0].dataset.filterByTagIds.split(",").map(id => parseInt(id)) : [];
```

Comment:
- A good occasion to use the `?.` operator.
- Don't save a variable that depends on what is in the DOM in `this`. You are just duplicating the information for no reason: this results in more code and potential bugs if that variable ever de-sync from what is actually in the DOM. Just use `this.$target[0].dataset.filterByTagIds` wherever you currenly use `this.tags`.